### DISCUSSION TITLE
What are some current thoughts on managing upstream keyword, reserved word, etc. changes?

### DISCUSSION THREAD

**Discussion by augb** [modular/modular#177] [Q&A] [answered]
Given the Mojo is a superset of Python 3, what are the current thoughts on how to manage changes to future versions of Python? For example, if Mojo uses the keyword `const` now, but sometime in the future the Python community decides to use `const`, as well. What sort of strategies or approaches might be taken to handle this? Would `def` functions be all about Python compatibility, but `fn` functions don't have to be? Are there other approaches such as allowing the aliasing of Python keywords (or vice versa) to work around this? Other thoughts?

**Comment by lattner** [comment, marked as answer]
For now, we need to get Mojo from 0.1 to at least 0.7 (conceptually, we have no specific versioning planned), at which point we'll understand more of what we're dealing with, and have broader developed relationships with the python community.

Also, my understanding is that Python3 generally doesn't take hard keywords for various compatibility reasons, even things like "case" are a soft keyword.    If that is true, then we may be fine.
